lib/count_words.rb

```diff
@@ -0,0 +1,23 @@
+class CountWords
+  attr_reader :counts, :total_word_count
+
+  def initialize
+    clear_counts
+    @total_word_count = 0
+  end
+
+  def clear_counts
+    @counts = Hash.new { |hash,key| hash[key] = 0 }
+  end
+
+  def count_words(input_string)
+    words = input_string.split(/[^-a-zA-Z']+/).map do |w|
```

Comment:
[String#scan](http://ruby-doc.org/core-2.0.0/String.html#method-i-scan) might make this a bit more readable.